### PR TITLE
Roq blog

### LINKED ISSUE #68: Rename integration-tests to roq-blog
The idea is:

> We will use our own blog as integration test :smile:

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #68

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Just looks like a minor Maven refactoring issue

```
Warning:  'parent.relativePath' of POM io.quarkiverse.roq:quarkus-roq-generator-spi:999-SNAPSHOT (/home/runner/work/quarkus-roq/quarkus-roq/roq-generator/spi/pom.xml) points at io.quarkiverse.roq:quarkus-roq-parent instead of io.quarkiverse.roq:quarkus-roq-generator-parent, please verify your project structure @ line 4, column 13
```

File: .github/workflows/deploy-blog.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          java-version: 17
+      - name: Build & Generate
+        working-directory: roq-blog
+        run: QUARKUS_ROQ_GENERATOR_BATCH=true ./mvnw -B package quarkus:run -Dquarkus.profile=gh-pages -DskipTests
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: roq-blog/target/roq
+  deploy:
+    if: github.repository_owner == 'ia3andy'
```
Comment (mathecruz) [inline .github/workflows/deploy-blog.yml]: Is it necessary @ia3andy?

File: common/deployment/src/main/java/io/quarkiverse/roq/deployment/RoqProjectProcessor.java [hunk 2]
```diff
@@ -17,19 +20,72 @@ public class RoqProjectProcessor {
...
+            OutputTargetBuildItem outputTarget) {
+        Path projectRoot = findProjectRoot(outputTarget.getOutputDirectory());
+        Path configuredSiteDirPath = Paths.get(config.siteDir().trim());
+        if (projectRoot == null || !Files.isDirectory(projectRoot)) {
+
+            if (configuredSiteDirPath.isAbsolute() && Files.isDirectory(configuredSiteDirPath)) {
+                configuredSiteDirPath = configuredSiteDirPath.normalize();
+            } else {
+                LOG.warn(
+                        "If not absolute, the Site directory is resolved relative to the project root, but Roq was not able to find the project root..");
```
Comment (mathecruz) [inline common/deployment/src/main/java/io/quarkiverse/roq/deployment/RoqProjectProcessor.java]:
```suggestion
                        "If not absolute, the Site directory is resolved relative to the project root, but Roq was not able to find the project root.");
```

File: roq-blog/site/_data/authors.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ia3andy:
+  name: "Andy Damevin"
+  email: "adamevin@redhat.com"
+  emailhash: "444d54a22db43a0a622180b1ca11dedb"
+  job_title: Principal Software Engineer
+  twitter: "ia3andy"
+  bio: "I am happy to wake up every day and make the Java world a better place with Quarkus. I am mostly a backend developer but I am not that bad at doing web apps either in my spare time, in fact, you can be the judge of that since Roq is my baby :-)"
```
Comment (mathecruz) [inline roq-blog/site/_data/authors.yml:7]: Nice bio <3 I loved it

File: roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/RoqFrontMatterProcessor.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package io.quarkiverse.roq.frontmatter.deployment;
 
 import static io.quarkiverse.roq.frontmatter.runtime.Page.LINK_KEY;
+import static io.quarkiverse.roq.util.PathUtils.addTrailingSlash;
 import static io.quarkiverse.roq.util.PathUtils.removeExtension;
 
 import java.util.*;
```
Comment (mathecruz) [inline roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/RoqFrontMatterProcessor.java:7]: Remove `.*`

File: roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/RoqFrontMatterProcessor.java [hunk 5]
```diff
@@ -17,6 +18,7 @@
 import io.quarkiverse.roq.frontmatter.deployment.items.RoqFrontMatterBuildItem;
 import io.quarkiverse.roq.frontmatter.deployment.items.RoqFrontMatterOutputBuildItem;
 import io.quarkiverse.roq.frontmatter.runtime.*;
```
Comment (mathecruz) [inline roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/RoqFrontMatterProcessor.java:20]: Remove `.*`

**Review by mcruzdev — APPROVED** [review]
I left a few comments @ia3andy, nice job :)

File: .github/workflows/deploy-blog.yml (hunk 1 above)
Comment (ia3andy) [reply .github/workflows/deploy-blog.yml]: Good catch

**Comment by ia3andy** [comment]
@mcruzdev when that works, could you in the deploy roq-blog, also generate the doc and put it in a doc directory

**Comment by melloware** [comment]
This is really fantastic.  When this is all done and we have a fancy UI i may try and switch my website blog to this from Gatsby.

**Comment by ia3andy** [comment]
> This is really fantastic. When this is all done and we have a fancy UI i may try and switch my website blog to this from Gatsby.

We still misses a few features (like pagination)

**Pull request merged**